### PR TITLE
Implement model input tools

### PR DESCRIPTION
**Pull request by kdeloach**
Serializes input UI values to client-side models.

To test:
1. Add scenario and change precipitation value
2. Create another scenario and add modifications / change precipitation value
3. Switching between scenarios should show correct user selected values

Connects #167

### CODE REVIEW

**Comment by kdeloach** [comment]
Rebased onto develop.

**Comment by kdeloach** [comment]
The controls are loaded dynamically now and may need design assistance.
image

File: src/mmw/js/src/modeling/controls.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+var LandCoverView = DrawControlView.extend({
+    template: landCoverTmpl,
+
+    getControlName: function() {
+        return 'landcover';
+    }
+});
+
+var ConservationPractiveView = DrawControlView.extend({
```
Comment (rajadain) [inline src/mmw/js/src/modeling/controls.js]: Typo, v -> c, should be `ConservationPracticeView`

**Comment by kdeloach** [comment]
@caseypt  Just noticed my changes don't strictly follow #165. I designed it such that there is no distinction between drawing/input modifications (other than 1 has a shape, and the other doesn't). This feature may need to be reworked.

**Comment by caseypt** [comment]
#165 was just a rough sketch, so it may not matter. The only difference between the two was that that inputs have a two-way relationship with scenarios, and modifications don't.

**Comment by kdeloach** [comment]
Rebased.

**Comment by lliss** [comment]
Missing a require statement for jQuery.

**Comment by lliss** [comment]
+1. Tested.

There's a lot of nesting of models and collections which makes things a little tricky to follow. But we've discussed that and I don't think we'd benefit from trying to do it another way.

**Comment by caseypt** [comment]
I'm not able to save models because `model_package` is invalid:

`{"model_package":["\"{u'name': u'tr-55', u'controls': [{u'name': u'landcover'}, {u'name': u'conservation_practice'}, {u'name': u'precipitation'}]}\" is not a valid choice."]}`

Under the current configuration, the backend only needs to know that it is `tr-55`.

**Comment by caseypt** [comment]
When loading saved projects, I get the following error:

selection_045

This is the line:

selection_046

**Comment by caseypt** [comment]
These errors are related. When `project.model_package`, when it comes back from the server is just a string equal to `tr-55`, which is why the error noted above occurs.

**Comment by caseypt** [comment]
These two tests are failing:

selection_047

Rebuilt the test bundle to make sure.

**Comment by caseypt** [comment]
@kdeloach There are some instructions for using the inspector to save a project in this PR comment: https://github.com/WikiWatershed/model-my-watershed/pull/230

**Comment by kdeloach** [comment]
Thanks, although I tested that saving works in #230, I have not made an effort to test things that have yet to be wired up to the UI. I do not really like using Marionette Inspector because of the double refresh problem.

**Comment by kdeloach** [comment]
Rebased.

**Comment by caseypt** [comment]
@kdeloach Is this ready for another run through?

**Comment by kdeloach** [comment]
Tests should be working now. Saving may be fixed by #278.

**Comment by lliss** [comment]
All right. Tried it out. All tests passing again.

**Pull request merged**